lib/faker/default/number.rb

```diff
@@ -139,6 +139,21 @@ def hexadecimal(legacy_digits = NOT_GIVEN, digits: 6)
         hex
       end
 
+      # Produces a number in binary format.
+      #
+      # @param digits [Integer] Number of digits in the he
+      # @return [String]
+      #
+      # @example
+      #   Faker::Number.binary(digits: 4) #=> "1001"
+      #
+      # @faker.version next
+      def binary(digits: 4)
+        bin = ''
+        digits.times { bin += rand(2).to_s(2) }
```

Comment:
This doesn't prevent the number from starting with a `0`, which means this could  produce `0010`, which technically has 4 digits, but in fact matches `10` which is 2 digits only.

Perhaps, it should always start with a 1?

```suggestion
        bin = '1'
        (digits - 1).times { bin += rand(2).to_s(2) }
```

Reply:
> This doesn't prevent the number from starting with a `0`, which means this could produce `0010`, which technically has 4 digits, but in fact matches `10` which is 2 digits only.
> 
> Perhaps, it should always start with a 1?

Hmm, Idk, I am thinking about use some option to choose, because I already worked in a simulator where this bit was important, so what do you think about it?

Reply:
I hadn’t thought of places where this could be important. I guess that it could stay as is, then, since having it always start with `1` also means that the range of possible values is more limited 👍